src/AspNet.Security.OAuth.Alipay/AlipayAuthenticationHandler.cs

```diff
@@ -0,0 +1,231 @@
+                ["sign_type"] = "RSA2",
+                ["version"] = "1.0",
+                ["auth_token"] = tokens.AccessToken,
+                ["timestamp"] = Clock.UtcNow.ToString("yyyy-MM-dd HH:mm:ss"),
+            };
+            sortedParams.Add("sign", GetRSA2Signature(sortedParams));
+
+            string address = QueryHelpers.AddQueryString(Options.UserInformationEndpoint, sortedParams);
+
+            using var response = await Backchannel.GetAsync(address);
+
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                /* Status: */ response.StatusCode,
+                                /* Headers: */ response.Headers.ToString(),
+                                /* Body: */ await response.Content.ReadAsStringAsync());
+
+                throw new HttpRequestException("An error occurred while retrieving user information.");
+            }
+
+            using var payload = await ParseJsonAsync(response, "alipay_user_info_share_response");
+            string statusCode = payload.RootElement.GetString("code");
+
+            if (statusCode != "10000")/* if code==10000 then success else failed. See: https://docs.open.alipay.com/common/105806 */
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                            "returned a {Status} response with the following message: {Message}.",
+                            /* Status: */ statusCode,
+                            /* Message: */ payload.RootElement.GetString("msg"));
+
+                throw new HttpRequestException("An error occurred while retrieving user information.");
+            }
+
+            identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, payload.RootElement.GetString("user_id"), ClaimValueTypes.String, Options.ClaimsIssuer));
+
+            var principal = new ClaimsPrincipal(identity);
+            var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, payload.RootElement);
+
+            context.RunClaimActions();
+
+            await Options.Events.CreatingTicket(context);
+            return new AuthenticationTicket(context.Principal, context.Properties, Scheme.Name);
+        }
+
+        /// <summary>
+        /// Gets an HTTP payload as an asynchronous operation.
+        /// </summary>
+        /// <param name="message">The message.</param>
+        /// <param name="elementName">Name of the element.</param>
+        /// <returns>Task&lt;JsonDocument&gt;.</returns>
+        private async Task<JsonDocument> ParseJsonAsync(HttpResponseMessage message, string elementName)
+        {
+            var messageStream = await message.Content.ReadAsStreamAsync();
```

Comment:
One question: are we 100% sure this provider ALWAYS returns UTF8-encoded payloads? Since their OAuth 2.0 is almost 100% wrong, I wouldn't be surprised if they returned weird charsets.

Reply:
Reqeust parameter ["charset"] = "UTF-8", so returns utf-8.